subprojects/platform-play/src/integTest/groovy/org/gradle/play/integtest/continuous/PlayReloadIntegrationTest.groovy

```diff
@@ -196,11 +204,130 @@ task otherTask {
         appIsRunningAndDeployed()
 
         when:
-        addHelloWorld()
+        addNewRoute()
 
         then:
         fails()
         !executedTasks.contains('runPlayBinary')
         errorPageHasTaskFailure("otherTask")
     }
+
+    void addPendingChangesHook() {
+        buildFile << """
+            ext.pendingChanges = new java.util.concurrent.atomic.AtomicBoolean(false)
+
+            void addPendingChangeListener() {
+                def pendingChangesManager = gradle.services.get(${PendingChangesManager.canonicalName})
+                pendingChangesManager.addListener new ${SingleFirePendingChangesListener.canonicalName}({
+                    synchronized(pendingChanges) {
+                        println "Pending changes detected"
+                        pendingChanges.set(true)
+                        pendingChanges.notifyAll()
+                    }
+                } as ${PendingChangesListener.canonicalName})
+            }
+
+            void waitForPendingChanges() {
+                synchronized(pendingChanges) {
+                    while(!pendingChanges.get()) {
+                        pendingChanges.wait()
+                    }
+                }
+            }
+        """
+    }
+
+    def "wait for changes to be built when a request comes in during initial app startup and there are pending changes"() {
```

Comment:
There's not really a way to block the app without the app running. Either we'd need to start a dummy Play app very early (and replace it later) or at least pretend to be the Play app until the real app is started.

This test is starting the app, pausing the build, immediately changing something, waiting for the change to be seen and then trying to access the app.